File: quantum-safe/postQuantum.js
// Quantum-safe NIST PQC: ML-KEM-768 (FIPS 203) and ML-DSA-65 (FIPS 204).
// npm i @noble/post-quantum
'use strict';
const { ml_kem768 } = require('@noble/post-quantum/ml-kem');
const { ml_dsa65 } = require('@noble/post-quantum/ml-dsa');

function mlkem768() {
  const { publicKey, secretKey } = ml_kem768.keygen();
  const { cipherText, sharedSecret } = ml_kem768.encapsulate(publicKey);
  const shared2 = ml_kem768.decapsulate(cipherText, secretKey);
  return Buffer.from(sharedSecret).equals(Buffer.from(shared2));
}

function mldsa65(message) {
  const { publicKey, secretKey } = ml_dsa65.keygen();
  const signature = ml_dsa65.sign(secretKey, message);
  return ml_dsa65.verify(publicKey, message, signature);
}

console.log('ML-KEM-768 shared-secret match:', mlkem768());
console.log('ML-DSA-65 verified:', mldsa65(new TextEncoder().encode('message')));
module.exports = { mlkem768, mldsa65 };

File: quantum-safe/post_quantum.rs
// Quantum-safe NIST PQC: ML-KEM-768 (FIPS 203) and ML-DSA-65 (FIPS 204).
// crates: pqcrypto-mlkem, pqcrypto-mldsa, pqcrypto-traits
use pqcrypto_mldsa::mldsa65;
use pqcrypto_mlkem::mlkem768;
use pqcrypto_traits::kem::SharedSecret;

pub fn mlkem768_kem() -> bool {
    let (public_key, secret_key) = mlkem768::keypair();
    let (shared1, ciphertext) = mlkem768::encapsulate(&public_key);
    let shared2 = mlkem768::decapsulate(&ciphertext, &secret_key);
    shared1.as_bytes() == shared2.as_bytes()
}

pub fn mldsa65_sign(message: &[u8]) -> bool {
    let (public_key, secret_key) = mldsa65::keypair();
    let signed = mldsa65::sign(message, &secret_key);
    mldsa65::open(&signed, &public_key).is_ok()
}
